Author: Maks Orlovich <morlovich@google.com>
Description: Add #include <zlib.h> when needed
--- a/src/net/instaweb/rewriter/image.cc
+++ b/src/net/instaweb/rewriter/image.cc
@@ -56,6 +56,8 @@
 #else
 #include "third_party/libpng/src/png.h"
 #endif
+
+#include <zlib.h>
 }
 
 using pagespeed::image_compression::AnalyzeImage;
--- a/src/pagespeed/kernel/image/image_converter.cc
+++ b/src/pagespeed/kernel/image/image_converter.cc
@@ -30,6 +30,9 @@
 #else
 #include "third_party/libpng/src/png.h"
 #endif
+
+#include <zlib.h>
+
 }  // extern "C"
 
 #include "base/logging.h"
--- a/src/pagespeed/kernel/image/png_optimizer_test.cc
+++ b/src/pagespeed/kernel/image/png_optimizer_test.cc
@@ -37,6 +37,9 @@
 #else
 #include "third_party/libpng/src/png.h"
 #endif
+
+#include <zlib.h>
+
 }
 
 namespace {
