Author: Maks Orlovich <morlovich@google.com>
Description: Convert default pagespeed.conf to new-style ACLs
--- a/src/install/common/pagespeed.conf.template
+++ b/src/install/common/pagespeed.conf.template
@@ -318,23 +318,12 @@
     # ModPagespeedUsePerVHostStatistics is enabled.  In that case,
     # /pagespeed_global_admin sees aggregated data across all vhosts,
     # and the /pagespeed_admin sees data only for a particular vhost.
-    #
-    # You may insert other "Allow from" lines to add hosts you want to
-    # allow to look at generated statistics.  Another possibility is
-    # to comment out the "Order" and "Allow" options from the config
-    # file, to allow any client that can reach your server to access
-    # and change server state, such as statistics, caches, and
-    # messages.  This might be appropriate in an experimental setup.
     <Location /pagespeed_admin>
-        Order allow,deny
-        Allow from localhost
-        Allow from 127.0.0.1
+        Require local
         SetHandler pagespeed_admin
     </Location>
     <Location /pagespeed_global_admin>
-        Order allow,deny
-        Allow from localhost
-        Allow from 127.0.0.1
+        Require local
         SetHandler pagespeed_global_admin
     </Location>
 
